### ISSUE TITLE
Portable PDB MethodDebugInformation table size differs from MethodDef

### ISSUE THREAD

**Issue by auduchinok** [Microsoft/visualfsharp#3514]
According to [Portable PDB v1.0 specs](https://github.com/dotnet/corefx/blob/master/src/System.Reflection.Metadata/specs/PortablePdb-Metadata.md#methoddebuginformation-table-0x31) MethodDebugInformation table should be either empty (missing) or have exactly as many rows as MethodDef table. In some situations F# compiler produces PDB that violates this rule.
A minimal repro solution: [Library.zip](https://github.com/Microsoft/visualfsharp/files/1257146/Library.zip).
Build log: https://gist.github.com/auduchinok/a622289a34db09066946e489593bcf09

Example:
```fsharp
namespace Ns

[<AbstractClass>]
type A() =
    abstract member M: int -> unit

type B() =
    inherit A()
    override x.M y = ()
```
MethodDebugInformation table doesn't contain a row for `Ns.B.M`.

**Comment by KevinRansom** [comment]
I will take a look, thanks for the report.

**Comment by KevinRansom** [comment]
@dsyme 
*  It's not urgency now Don ... unless you can point to something that's broken.  
*  Clearly it isn't a regression since portable PDBS are new.

**Comment by dsyme** [comment]
@KevinRansom Ha, didn't even realise you'd removed the label :)  Just saw it was without an urgency.  Make it Later if it's really unimportant

**Comment by dsyme** [comment]
@auduchinok Do you know if this causes any known problems for any consuming tools?

**Comment by auduchinok** [comment]
I had had exceptions in dotPeek and ILSpy and the Rider team had to remove an assertion in debugger.
I don't know about other tools but I guess there may assertions checking this rule somewhere as well.

**Comment by KevinRansom** [comment]
@auduchinok.   I hope to look at this over the weekend.

**Comment by auduchinok** [comment]
It is works now in master, thank you @KevinRansom.
It would be great to include the fix into the next dotnet SDK update.

**Comment by cartermp** [comment]
@livarcocc, we recently fixed a bug in the compiler that affects debugging with portable PDBs.  Which branch should we target for an update?

**Comment by livarcocc** [comment]
We are creating a release/15.5 branch today. It should target that.

**Comment by auduchinok** [comment]
@dsyme @cartermp Just to add another case: I got ReSharperHost crashing on Mono when loading a R# plugin and FCS compiled using compilers from installed toolsets (Mono 5.2, dotnet 2.0), and it works fine using fsc.exe built from master.

**Comment by dsyme** [comment]
Was fixed in #3514

@auduchinok If you still have a repro for the 2nd case you mentioned above please enter it as a separate bug
